rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -44,33 +61,87 @@
  */
 public class FlowNodeUtil {
 
+    protected static final Cache<String,List<FlowNode>> executionCache = CacheBuilder.newBuilder().weakValues().maximumSize(100).build();
```

Comment:
@jglick Ah, okay it did update with latest commit.  In any case, the executionCache only needs to live as long as the current run is being actively analyzed and no longer.  One alternative would be to use a very small cache size but  strong values, assuming holding references to a few WorkflowJobs is not the end of the world.